modules/core/src/main/java/org/apache/synapse/mediators/elementary/Target.java

```diff
@@ -527,9 +527,12 @@ private String removeJSONFromString(MessageContext synCtx, String inputString, S
                 if (path.equals("$") || path.equals("$.")) {
                     result = "";
                 } else {
-                    DocumentContext doc = JsonPath.parse(result);
-                    doc.delete(path);
-                    result = doc.jsonString();
+                    Object list = JsonPath.compile(path).read(result);
```

Comment:
shall we add an info log saying the filter gave an empty result, so we are not performing the delete.

Reply:
Updated